recipe/meta.yaml

```diff
@@ -18,18 +18,22 @@ build:
 
 requirements:
   host:
-    - python >=3.7
+    - python >=3.7,<4.0
     - pip
     - poetry-core >=1.1.0a6
   run:
-    - python >=3.7
+    - python >=3.7,<4.0
     - attrs >=20
+    # these are from extras
+    - exceptiongroup
     - typing_extensions
```

Comment:
@Tinche, these are listed as extras in the upstream package:
https://github.com/python-attrs/cattrs/blob/main/pyproject.toml#L26-L27
but imports failed here without them.  Maybe they're not extras after all?

Reply:
@xylar They're not extras, they're just conditional. Absolutely required depending on the Python version.

Reply:
If you download the pypi package, they are listed as extras: https://files.pythonhosted.org/packages/f7/06/9d81ad262e0146c0f2e94c6c450f6c4c490ce2e4711f30e546bb04883b62/cattrs-22.1.0.tar.gz

Reply:
```
install_requires = \
['attrs>=20']

extras_require = \
{':python_version <= "3.10"': ['exceptiongroup'],
 ':python_version >= "3.7" and python_version < "3.8"': ['typing_extensions']}
```
I don't know enough about how a `pyproject.toml` gets converted to a `setup.py` in the pypi package but it seems like something is misconfigured or buggy.